editors/vscode/src/quick_picks.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
+
+import * as vscode from "vscode";
+import * as fs from "fs-extra";
+import simpleGit from "simple-git";
+import * as path from "node:path";
+
+export async function newProject(context: vscode.ExtensionContext) {
+    type Language = "Node (JavaScript/TypeScript)" | "c++" | "rust";
+
+    const language = await vscode.window.showQuickPick(
+        ["Node (JavaScript/TypeScript)", "c++", "rust"],
+        {
+            placeHolder: "What language do you want to use?",
+        },
+    );
+
+    if (!language) {
+        vscode.window.showErrorMessage("Language selection is required.");
+        return;
+    }
+
+    let repoUrl: string;
+    switch (language as Language) {
+        case "Node (JavaScript/TypeScript)":
+            repoUrl = "https://github.com/slint-ui/slint-nodejs-template";
+            break;
+        case "c++":
+            repoUrl = "https://github.com/slint-ui/slint-cpp-template";
+            break;
+        case "rust":
+            repoUrl = "https://github.com/slint-ui/slint-rust-template";
```

Comment:
unfortunately, the rust-template is not usable as such because it is templated through cargo-generate and we need to replace things like this:
https://github.com/slint-ui/slint-rust-template/blob/f2508d37719e51a41dca994105ca3cfbb519ef2f/Cargo.toml#L2-L4
(I think that's the only things that needs replacement so it could be done in JS to search and replace this

Reply:
Were you thinking if the user selects Rust then we ask for the author name? The flow already asks for the project name

Reply:
we could also remove the `author=` from the template, this is not anymore mandatory field.

Reply:
Ah even better. I will do this and then maybe we go with that? Then other tweaks can be added tomorrow.